prow/plugins/blunderbuss/blunderbuss.go

```diff
@@ -272,14 +279,20 @@ func getReviewers(rc reviewersClient, ghc githubClient, log *logrus.Entry, autho
 		}
 		leafReviewers = leafReviewers.Union(fileUnusedLeafs)
 		if r := findReviewer(ghc, log, useStatusAvailability, &busyReviewers, &fileUnusedLeafs); r != "" {
-			reviewers.Insert(r)
+			if !reviewers.Has(r) {
+				reviewersList = append(reviewersList, r)
+				reviewers.Insert(r)
+			}
 		}
 	}
 	// now ensure that we request review from at least minReviewers reviewers. Favor leaf reviewers.
```

Comment:
@fejta see here it even says it will favor leaf reviewers...

Reply:
Yes we should favor leaf reviewers, but we shouldn't prioritize inside the leafs reviewers

Reply:
randomization is now done just before return